### DISCUSSION TITLE
Using SCIE for Shadow register?

### DISCUSSION THREAD

**Discussion by yathivunnam** [chipsalliance/rocket-chip#3030] [Ideas]
Can we use SCIE (Simple Custom Instruction Extension) to implement a shadow register? I see that the SCIE has two dedicated registers to store data and one register to output  data. I want to implement a shadow register which just stores data from registers. So that it can be compared.

**Comment by jerryz123** [comment]
Instructions are sent to the SCIE unit speculatively, and no signals are provided to indicated if a SCIE instruction has committed.
Therefore, it is unsafe to maintain architectural state in a SCIE extenstion.

The RoCC interface is likely more appropriate for maintaining a shadow copy of architectural registers.
